array/array.mbt

```diff
@@ -76,11 +70,44 @@ test "iteri" {
       i = i + 1
     },
   )
-  if failed {
-    return Err("iteri test failed")
+  @assertion.assert_false(failed)?
+}
+
+pub fn iter_rev[T](self : Array[T], f : (T) -> Unit) {
+  loop self.length() - 1 {
+    i => if i > 0 { f(self[i]); continue i - 1 }
```

Comment:
same here using for :  `for i = self.length() - 1; i >= 0 ; i = i - 1 {...}` check the boundary to see if `i=0` is applied